core/payment/src/processor.rs

```diff
@@ -1,3 +1,4 @@
+use crate::dao;
```

Comment:
I dont think this import is needed, it is actually confusing to me:
- all other dao's are imported by name
- in this file `let dao` is used pretty often

Would remove this import and use (already imported) `AllocationDao` instead